### PR TITLE
Components + sessions: calling APIs internally

### PR DESCRIPTION
**Pull request by jreij**
## Description
Initial implementation of components in sessions. In this branch we added the ability to initialize the Card Component with a session, and the logic to internally call the session APIs.

## Checklist 
- [x] Changes are tested manually

COAND-699

### CODE REVIEW

File: card/src/main/java/com/adyen/checkout/card/CardComponentProvider.kt [hunk 1]
```diff
@@ -101,11 +109,97 @@ class CardComponentProvider(
...
                 genericActionDelegate = genericActionDelegate,
                 actionHandlingComponent = DefaultActionHandlingComponent(genericActionDelegate, cardDelegate),
+                savedStateHandle = savedStateHandle,
             )
         }
         return ViewModelProvider(viewModelStoreOwner, factory)[key, CardComponent::class.java]
     }
 
+    @Suppress("LongParameterList", "LongMethod")
+    fun get(
```
Comment (OscarSpruit) [inline card/src/main/java/com/adyen/checkout/card/CardComponentProvider.kt:119]: The original `get` method and this one share a lot of code, we should see if we can extract the similar things

Comment (OscarSpruit) [reply card/src/main/java/com/adyen/checkout/card/CardComponentProvider.kt]: Addressed it in 5e47fd8ecde09fca462588bdb0169c71c19d49e1

File: sessions-core/src/main/java/com/adyen/checkout/sessions/CheckoutSession.kt [hunk 2]
```diff
@@ -8,9 +8,16 @@
...
 
+import com.adyen.checkout.components.model.paymentmethods.PaymentMethod
 import com.adyen.checkout.sessions.model.setup.SessionSetupResponse
 
 // TODO docs
 data class CheckoutSession(
     val sessionSetupResponse: SessionSetupResponse
-)
+) {
+    fun getPaymentMethod(paymentMethodType: String): PaymentMethod? {
```
Comment (ozgur00) [inline sessions-core/src/main/java/com/adyen/checkout/sessions/CheckoutSession.kt:18]: Shall we add this helper for stored payment methods too?
```kotlin
    fun getStoredPaymentMethod(predicate: (StoredPaymentMethod) -> Boolean): StoredPaymentMethod? {
        return sessionSetupResponse.paymentMethods?.storedPaymentMethods.orEmpty().firstOrNull {
            predicate(it)
        }
    }
```

Comment (OscarSpruit) [reply sessions-core/src/main/java/com/adyen/checkout/sessions/CheckoutSession.kt:18]: Why do we need the predicate again?

Comment (ozgur00) [reply sessions-core/src/main/java/com/adyen/checkout/sessions/CheckoutSession.kt:18]: For stored payment methods we don't know what they're gonna use to choose the payment method they want. I think this is the best option we can provide for that so they can pass the condition they desire.

**Review by ozgur00 — APPROVED** [review]

**Pull request merged**